concurrent_test.go

```diff
@@ -347,27 +347,47 @@ func (w *worker) name() string {
 
 func (w *worker) run() (historyRecords, error) {
 	var rs historyRecords
+
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
 	for {
 		select {
 		case <-w.stopCh:
-			w.t.Logf("%q finished.", w.name())
 			return rs, nil
 		default:
 		}
 
-		op := w.pickOperation()
-		bucket, key := w.pickBucket(), w.pickKey()
-		rec, err := executeOperation(op, w.db, bucket, key, w.conf)
-		if err != nil {
-			readErr := fmt.Errorf("[%s: %s]: %w", w.name(), op, err)
-			w.t.Error(readErr)
-			w.errCh <- readErr
-			return rs, readErr
-		}
+		err := w.db.Update(func(tx *bolt.Tx) error {
+		inner:
+			for {
+				op := w.pickOperation()
+				bucket, key := w.pickBucket(), w.pickKey()
+				rec, eerr := executeOperation(op, tx, bucket, key, w.conf)
+				if eerr != nil {
+					opErr := fmt.Errorf("[%s: %s]: %w", w.name(), op, eerr)
+					w.t.Error(opErr)
+					w.errCh <- opErr
+					return opErr
+				}
+
+				rs = append(rs, rec)
+				if w.conf.workInterval != (duration{}) {
+					time.Sleep(randomDurationInRange(w.conf.workInterval.min, w.conf.workInterval.max))
+				}
 
-		rs = append(rs, rec)
-		if w.conf.workInterval != (duration{}) {
-			time.Sleep(randomDurationInRange(w.conf.workInterval.min, w.conf.workInterval.max))
+				select {
+				case <-ticker.C:
+					break inner
```

Comment:
I think we can just return nil here.

Reply:
Yes, you are right. 

We can also remove the label "break inner" syntax @tjungblu

Reply:
updated.